src/DecoderPlugin.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Client\Exception;
+use Http\Encoding\DechunkStream;
+use Http\Encoding\DecompressStream;
+use Http\Encoding\GzipDecodeStream;
+use Http\Encoding\InflateStream;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allow to decode response body with a chunk, deflate, compress or gzip encoding
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class DecoderPlugin implements Plugin
+{
+    /**
+     * @var bool Whether this plugin decode stream with value in the Content-Encoding header (default to true).
+     *
+     * If set to false only the Transfer-Encoding header will be used.
+     */
+    private $useContentEncoding;
+
+    public function __construct($useContentEncoding = true)
```

Comment:
i think we should have the long description of useContentEncoding as docblock on the constructor as this is what people doing `new DecoderPlugin` will see.